Dockerfiles/agent/datadog-k8s-docker.yaml

```diff
@@ -1,22 +0,0 @@
-## Provides autodetected defaults, for kubernetes environments
-## where configuration templates are set as docker labels
-## please see datadog.yaml.example for all supported options
-
-# Autodiscovery for Kubernetes
-listeners:
-  - name: kubelet
-config_providers:
-  - name: kubelet
-    polling: true
-  # needed to support legacy docker label config templates
-  - name: docker
-    polling: true
-    poll_interval: 1s
-
-apm_config:
-  apm_non_local_traffic: true
-  max_memory: 0
-  max_cpu_percent: 0
-
-# Use java container support
-jmx_use_container_support: true
```

Comment:
why do we remove these from `datadog-k8s-docker.yaml` and we keep them in `datadog-kubernetes.yaml` ?
as opposed to `listeners` and `config_providers`, jmx and apm config are mapped to env vars, IMHO we should remove these files and use env vars instead.
(cc @clamoriniere related to what we discussed earlier)

Reply:
ok in `Dockerfiles/agent/entrypoint/50-kubernetes.sh`  I see why you removed `datadog-k8s-docker.yaml`, any thoughts on using env vars for apm and jmx config?

Reply:
It's an option but it will only work with new deployments of our Helm chart or Operator, it may break if customer upgrades Agent version only (or use "latest" tag).

Reply:
I'm thinking we can set them in the entrypoint script

Reply:
We don't run "entrypoint" scripts in K8S, we run the init script in an init-container then the Agent container runs directly Agent binary (except on Windows), so entrypoint cannot modify environment.